client/components/TestRenderer/index.jsx

```diff
@@ -217,100 +218,182 @@ const ErrorComponent = ({ hasErrors = false }) => {
-     * is triggered.
-     *
-     * This is constructing the 'scripts' STRING variable as an internal object,
-     * '_scripts' if there are an available 'scripts' variable using eval.
-     */
-    let _scripts = null;
-    const _scriptsString = scripts.replace('var scripts', '_scripts');
-    eval(_scriptsString);
-
-    const testRunIO = new TestRunInputOutput();
-    testRunIO.setTitleInputFromTitle(title);
-    testRunIO.setUnexpectedInputFromBuiltin();
-    testRunIO.setScriptsInputFromMap(
-        typeof _scripts === 'object' ? _scripts : {}
-    );
-    testRunIO.setSupportInputFromJSON(support);
-    testRunIO.setConfigInputFromQueryParamsAndSupport(configQueryParams); // Array.from(new URL(document.location).searchParams)
-    testRunIO.setKeysInputFromBuiltinAndConfig();
-    testRunIO.setCommandsInputFromJSONAndConfigKeys(commandJson);
-    testRunIO.setBehaviorInputFromJSONAndCommandsConfigKeysTitleUnexpected(
-        testJson
-    );
-    testRunIO.setPageUriInputFromPageUri(testPageUri);
+    const [state, setState] = useState(null);
+    const [submitResult, setSubmitResult] = useState(null);
+    const [submitCalled, setSubmitCalled] = useState(false);
+
+    const setup = async () => {
+        const testRunIO = new TestRunInputOutput();
+
+        // Array.from(new URL(document.location).searchParams)
+        const configQueryParams = [['at', evaluateAtNameKey(at.name)]];
+
+        await testRunIO.setInputsFromCollectedTestAsync(renderableContent);
+        testRunIO.setConfigInputFromQueryParamsAndSupport(configQueryParams);
+        testRunIO.setPageUriInputFromPageUri(testPageUrl);
+
+        const _state = mergeState(testRunIO.testRunState(), scenarioResults);
+
+        const testWindow = new TestWindow({
+            ...testRunIO.testWindowOptions(),
+            hooks: {
+                windowOpened() {
+                    testRunExport.dispatch(userOpenWindow());
+                },
+                windowClosed() {
+                    testRunExport.dispatch(userCloseWindow());
+                }
+            }
+        });
 
-    const testWindow = new TestWindow({
-        ...testRunIO.testWindowOptions(),
-        hooks: {
-            windowOpened() {
-                testRunExport.dispatch(userOpenWindow());
+        const testRunExport = new TestRunExport({
+            hooks: {
+                openTestPage() {
+                    testWindow.open();
+                },
+                closeTestPage() {
+                    testWindow.close();
+                }
             },
-            windowClosed() {
-                testRunExport.dispatch(userCloseWindow());
-            }
+            resultsJSON: state => testRunIO.submitResultsJSON(state),
+            state: _state
+        });
+        setState(_state);
+        setTestRunExport(testRunExport);
+    };
+
+    const mergeState = (state, scenarioResults = []) => {
```

Comment:
This function is where the magic happens ✨ I'm very thankful that you figured this one out.

To me the word "remapping" would make sense to include in the function name since it's remapping the modeling from the GraphQL schema to the TestRenderer schema. My instinct is that I wouldn't necessarily call it a merge because it's not just combining things, it's also translating them from one universe to another.

Reply:
> To me the word "remapping" would make sense to include in the function name since it's remapping the modeling from the GraphQL schema to the TestRenderer schema. My instinct is that I wouldn't necessarily call it a merge because it's not just combining things, it's also translating them from one universe to another.

Fair point, will revise